lib/smallvector.h

```diff
@@ -41,6 +41,15 @@ struct TaggedAllocator : std::allocator<T>
     TaggedAllocator(Ts&&... ts)
         : std::allocator<T>(std::forward<Ts>(ts)...)
     {}
+
+    template<class U>
+    TaggedAllocator(const TaggedAllocator<U, N>);
```

Comment:
this constructor has no implementation right? Should it be marked with `=delete`?

Reply:
I can't test it but I wonder if this constructor should be removed and let the other constructor be used instead. What happens if you comment out line 45 and 46 and recompile?